### PR TITLE
SMT model is sorted and printed as secondary location

### LINKED ISSUE #4647: SMT: change value into <result> or something better
```
nonlinear.sol:7:12: Warning: Overflow (resulting value larger than 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff) happens here
        uint c = a * b;
                 ^---^
  for:
  a = 2
  b = 0x8000000000000000000000000000000000000000000000000000000000000000
  c = 0
  value = 0x010000000000000000000000000000000000000000000000000000000000000000
```

In this example `value` corresponds to the unlimited precision result of the operation, which is then trunacted to the range of the output (in this case `c`)

I think `<result>` would be an appropriate replacement, especially as that is an invalid variable name and wouldn't be confused as a variable.

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes #4647.

### CODE REVIEW

**Review by ekpyron — DISMISSED** [review]

File: libsolidity/formal/SMTChecker.cpp [hunk 1]
```diff
@@ -597,19 +597,23 @@ void SMTChecker::checkCondition(
...
-					message << "  " << expressionNames.at(i) << " = " << values.at(i) << "\n";
+					sortedModel[expressionNames.at(i)] = values.at(i);
+
+			for (auto const& eval: sortedModel)
+				modelMessage << "  " << eval.first << " = " << eval.second << "\n";
 		}
 		else
 			message << ".";
-		m_errorReporter.warning(_location, message.str() + loopComment);
+		m_errorReporter.warning(_location, message.str() + loopComment, SecondarySourceLocation().append(modelMessage.str(), SourceLocation()));
```
Comment (ekpyron) [inline libsolidity/formal/SMTChecker.cpp]: I just approved, but actually I think it would be cleaner to declare ``SecondarySourceLocation ssl;`` in the beginning of the scope and to only do the ``append`` inside the ``if``. ``modelMessage`` could be local to the ``if`` scope as well then.

Otherwise we end up with empty secondary source locations being added if ``!m_currentFunction``.

File: libsolidity/formal/SMTChecker.cpp [hunk 2]
```diff
@@ -252,14 +252,14 @@ void SMTChecker::checkUnderOverflow(smt::Expression _value, IntegerType const& _
 		_value < SymbolicIntVariable::minValue(_type),
 		_location,
 		"Underflow (resulting value less than " + formatNumber(_type.minValue()) + ")",
-		"value",
+		"<result>",
```
Comment (axic) [inline libsolidity/formal/SMTChecker.cpp:255]: Can you split this into a separate commit?

**Review by ekpyron — DISMISSED** [review]
Looks fine to me (split or unsplit).

File: libsolidity/formal/SMTChecker.cpp (hunk 2 above)
Comment (leonardoalt) [reply libsolidity/formal/SMTChecker.cpp:255]: Done

**Review by axic — APPROVED** [review]
